src/core/backbone_router/bbr_manager.cpp

```diff
@@ -118,6 +126,99 @@ void Manager::SendMulticastListenerRegistrationResponse(const Coap::Message &
     otLogInfoBbr("Sent MLR.rsp (status=%d): %s", aStatus, otThreadErrorToString(error));
 }
 
+void Manager::HandleDuaRegistration(const Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
+{
+    otError                    error               = OT_ERROR_NONE;
+    ThreadStatusTlv::DuaStatus status              = ThreadStatusTlv::kDuaSuccess;
+    bool                       isPrimary           = Get<BackboneRouter::Local>().IsPrimary();
+    uint32_t                   lastTransactionTime = 0;
+    Ip6::Address               target;
+    Ip6::InterfaceIdentifier   meshLocalIid;
+
+    VerifyOrExit(aMessage.IsConfirmable() && aMessage.GetCode() == OT_COAP_CODE_POST, error = OT_ERROR_PARSE);
+
+    SuccessOrExit(error = Tlv::FindTlv(aMessage, ThreadTlv::kTarget, &target, sizeof(target)));
+    SuccessOrExit(error = Tlv::FindTlv(aMessage, ThreadTlv::kMeshLocalEid, &meshLocalIid, sizeof(meshLocalIid)));
+
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    if (mDuaResponseIsSpecified)
+    {
+        if (mDuaResponseTargetMlIid.IsUnspecified() || mDuaResponseTargetMlIid == meshLocalIid)
```

Comment:
Can we combine the two `if`()` into one?
```suggestion
    if (mDuaResponseIsSpecified && ((mDuaResponseTargetMlIid.IsUnspecified() || mDuaResponseTargetMlIid == meshLocalIid))
```